WatsonTcp/WatsonTcpServer.cs

```diff
@@ -1411,7 +1418,7 @@ private void SendDataStream(ClientMetadata client, long contentLength, Stream st
 
             long bytesRemaining = contentLength;
             int bytesRead = 0;
-            byte[] buffer = new byte[_Settings.StreamBufferSize];
+            byte[] buffer = _BufferPool;

```

Comment:
This will not be safe as a server instance property.  It would need to be a property of ```ClientMetadata``` instead.

Reply:
FYI I see what you're doing here and am implementing on my end, will credit you in the commit!

Reply:
NuGet: https://www.nuget.org/packages/WatsonTcp/4.8.14.6
Commit: https://github.com/jchristn/WatsonTcp/commit/ecf95422d1a537f5d2d1cf11e489b9c3db86bc99

Credited you with the enhancement.  Thanks for putting this together @Ahmed310